addons/mail/models/mail_channel.py

```diff
@@ -746,6 +746,41 @@ def _channel_message_notifications(self, message, message_format=False):
     #   - when a message is posted on a channel (to the channel, using _notify() method)
     # ------------------------------------------------------------
 
+    def set_message_pin(self, message_id, pinned):
+        """ (Un)pin a message on the channel and send a notification to the
+        members.
+        :param message_id: id of the message to be pinned.
+        :param pinned: whether the message should be pinned or unpinned.
+        """
+        self.ensure_one()
+        message_to_update = self.message_ids.filtered(
+            lambda msg: msg.id == message_id and bool(msg.pinned_at) != pinned
+        )
+        if not message_to_update:
+            return
+        message_to_update.write({'pinned_at': fields.datetime.now() if pinned else False})
+        self.env['bus.bus']._sendone(self, 'mail.record/insert', {
+            'Message': {
+                'id': message_id,
+                'pinned_at': fields.Datetime.to_string(message_to_update.pinned_at),
+            }
+        })
+        if pinned:
+            notification_text = f'''
+                <div data-oe-type="pin" class="o_mail_notification">
+                    %(user_name_pinned_a)s <a href="#" data-oe-type="highlight" data-oe-id="{message_id}">%(message)s </a>%(to_this_channel)s
+                    <a href="#" data-oe-type="pin-menu">%(see_all_pins)s
+                    </a>
+                </div>
+            '''
+            notification = Markup(notification_text) % {
+                'user_name_pinned_a': _('%s pinned a' % self.env.user.display_name),
+                'message': _('message'),
+                'to_this_channel': _('to this channel.'),
+                'see_all_pins': _('See all pinned messages.'),
+            }
```

Comment:
This format is really strange and it will lead to really unclear translation terms.
Separating "a" from "message" is not good.
Using `%` inside the translation `()` will not lead to expected result.

Suggestion: make the first phrase a single translation block (including few bits of html of necessary).
The 2nd phrase is ok.